Source/Core/CameraLimiter.js

```diff
@@ -0,0 +1,301 @@
+        var maximum = new Cartesian3(0.5, 0.5, 0.5);
+
+        var returningPosition = positionInObjectSpace;
+
+        returningPosition.x = Math.max(minimum.x, Math.min(maximum.x, returningPosition.x));
+        returningPosition.y = Math.max(minimum.y, Math.min(maximum.y, returningPosition.y));
+        returningPosition.z = Math.max(minimum.z, Math.min(maximum.z, returningPosition.z));
+
+        return Matrix3.multiplyByVector(Matrix3.inverse(inverseTransformationMatrix, new Matrix3.fromScale(1.0)), returningPosition, returningPosition);
+    };
+
+    /**
+     * If inputted orientation is valid within the min and max {@link HeadingPitchRoll} limits defined for this limiter, returns orientation.
+     * Otherwise returns closest valid orientation within the min and max {@link HeadingPitchRoll} limits defined for this limiter.
+     *
+     * @param {HeadingPitchRoll} orientation The orientation we are checking to be between the limiter's min and max {@link HeadingPitchRoll} attributes.
+     * @return {HeadingPitchRoll} A valid orientation.
+     */
+    CameraLimiter.prototype.closestOrientationTo = function(orientation) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('orientation', orientation);
+        if (!(orientation instanceof HeadingPitchRoll)) {
+            throw new DeveloperError('orientation required to be of type HeadingPitchRoll.');
+        }
+
+        // if min is defined, max must also be. if min is not defined, max must also not be.
+        if (!defined(this.minHeadingPitchRoll) && defined(this.maxHeadingPitchRoll)) {
+            throw new DeveloperError('minHeadingPitchRoll is required.');
+        }
+        if (defined(this.minHeadingPitchRoll) && !defined(this.maxHeadingPitchRoll)) {
+            throw new DeveloperError('maxHeadingPitchRoll is required.');
+        }
+        //>>includeEnd('debug');
+
+        if(!defined(this.minHeadingPitchRoll)) {
+            return orientation;
+        }
+
+        var returningOrientation = orientation;
+
+        returningOrientation.heading = Math.max(this.minHeadingPitchRoll.heading, Math.min(this.maxHeadingPitchRoll.heading, returningOrientation.heading));
+        returningOrientation.pitch = Math.max(this.minHeadingPitchRoll.pitch, Math.min(this.maxHeadingPitchRoll.pitch, returningOrientation.pitch));
+        returningOrientation.roll = Math.max(this.minHeadingPitchRoll.roll, Math.min(this.maxHeadingPitchRoll.roll, returningOrientation.roll));
+
+        return returningOrientation;
+    };
+
+    /**
+     * Duplicates this CameraLimiter instance.
+     *
+     * @param {CameraLimiter} [result] The object onto which to store the result.
+     * @returns {CameraLimiter} The modified result parameter or a new CameraLimiter instance if one was not provided.
+     */
+    CameraLimiter.clone = function(limiter, result) {
+        if (!defined(result)) {
+            var options = {};
+            result = new CameraLimiter(options);
```

Comment:
No need for the options parameter, you already set the options to a default value in the constructor.